pulp_2to3_migration/tests/functional/test_file_migration.py

```diff
@@ -126,3 +151,8 @@ def test_2_migrate_all_iso_repositories(self):
         """Test that a Migration Plan to mirror Pulp 2 executes correctly."""
         repos = list(PULP_2_ISO_FIXTURE_DATA.keys())
         self._do_test(repos, EMPTY_ISO_MIGRATION_PLAN)
+
+    def test_3_migrate_iso_repositories_with_different_importer(self):
+        """Test that a Migration Plan with different importers executes correctly."""
+        repos = list(PULP_2_ISO_FIXTURE_DATA.keys())
```

Comment:
Since the migration plan only specifies two repositories, you need to provide only those two repos here, as test 1 does.  That should fix the test failure when run individually.  However, we really need a mechanism to clear out the Pulp 3 database between test runs.

Reply:
Yes, that's the only reason there are all 4 specified at the moment.
It's still helpful to have it, having a task succeeded without failures is also a test which is better than nothing.
I tried to add checks for the importer but it would be specific to a plan and not as generic as you were able to implement the rest, so I gave up on adding it with this pr.
Until we introduce a cleanup mechanism, I'm not sure if there are options to improve the test. I can put it before "migrate-all" and then it will be two repos.